app/models/calagator/event/saver.rb

```diff
@@ -34,7 +34,17 @@ def attempt_save?
       end
 
       def spam?
-        evil_robot? || too_many_links?
+        evil_robot? || too_many_links? || ml_flags_as_spam?
+      end
+
+      def get_spam_label
+        GpturkService.get_spam_label("#{params.dig(:event, :title)} #{params.dig(:event, :description)} #{params[:start_date]} at #{params[:start_time]}")
+      end
+
+      def ml_flags_as_spam?
+        if get_spam_label != 1
+          self.failure = "<h3>Spammer</h3> We didn't save this event because we think this looks like a spammy event. If this isn't spam and is a legitimate event, please file a bug report and let us know."
```

Comment:
Thoughts on not including a failure message? What about shadow banning it?

Reply:
I feel like it's probably fair to let people know what's happening here? False positives should be rare, but if they happen, it may be valuable to be clear about steps for remediation?

Reply:
Yeah, I wouldn't want legit users to wonder what's up if they happen to get flagged.